### ISSUE TITLE
APSIM classic source code compile fail

### ISSUE THREAD

**Issue by JoannaYRS** [APSIMInitiative/APSIMClassic#2225]
Hi,

I'm currently facing some challenges while attempting to compile the entire APSIM model from its source code. Following the steps on the webpage, I encounter [Fail] returns, including '[Fail] Compile General' and '[Fail] Compile ShellExtensions'. I'm using Visual Studio 2017, and I've attached screenshots of the compilation printout for reference.

Additionally, while inspecting the Maize model's code, I noticed incomplete files. For instance, in 'Maize/plant.h', the following lines are present:

```cpp
#include <ComponentInterface2/ScienceAPI2.h>
#include <ComponentInterface2/DataTypes.h>
```
However, these two files are missing from the 'ComponentInterface2' directory.

I would appreciate any suggestions or answers. Thank you.

![MicrosoftTeams-image](https://github.com/APSIMInitiative/APSIMClassic/assets/132005487/9de49741-25c9-4e9a-a5a7-c8f1b8774676)
![MicrosoftTeams-image (1)](https://github.com/APSIMInitiative/APSIMClassic/assets/132005487/0e2124b3-c08f-4652-b176-edd22c69c31d)
![MicrosoftTeams-image (2)](https://github.com/APSIMInitiative/APSIMClassic/assets/132005487/15d3c672-ddad-4a2c-a545-47cb329b10d7)
![MicrosoftTeams-image (3)](https://github.com/APSIMInitiative/APSIMClassic/assets/132005487/bee90c91-077a-44bc-9c55-b2f5a718744d)

**Comment by peter-devoil** [comment]
> .. incomplete files. 

These files are generated by the build process; and if the build process fails, they won't be there. 

I'm guessing that the root issue is that it's not setting up the VS2017 build environment correctly - that's the batch file "VSDevCmd.bat" error. It relies on that to  to the autogenerating bit above.  Can you investigate if that file is installed, if not - did you install the 32 bit compiler?

**Comment by JoannaYRS** [comment]
Thank you very much for your response @peter-devoil . Upon inspecting the BuildAll.bat file, I discovered that the path to VsDevCmd.bat is incorrect for my situation:
```rem ----- Set up the Visual Studio compiler tools
call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"
```
This is because I originally used Professional 2017.  To avoid potential errors, I reinstalled Visual Studio 2017 Community and recompiled APSIM. However, I encountered a new error.
I appreciate any assistance in resolving this issue. Thank you!
![image](https://github.com/APSIMInitiative/APSIMClassic/assets/132005487/9a59c7f1-cf88-425d-810a-3d555dc0f2a3)

**Comment by peter-devoil** [comment]
You can ignore the cotton error if cotton isn't in your simulations

ShellExtensions are likewise for windows XP - anachronisms.

The missing xml libraries are usually installed alongside the build tree - that's step 2 in https://www.apsim.info/documentation/model-documentation/developer-documentation/building-apsim-from-source/.

I just checked whether that BuildLibraries archive exists - and it appears to have moved. @hol353 - any clues where?

**Comment by peter-devoil** [comment]
@JoannaYRS - drop me an email at peter dot devoil at uq.edu.au and I'll send you a download link

@hol353 - perhaps that .7z archive should be in the repo so we dont keep losing it..

**Comment by hol353** [comment]
Yes I think the archive should be in the repo somewhere. Does someone want to raise a PR and add it?

**Comment by JoannaYRS** [comment]
Thank you very much @peter-devoil  and @hol353 
Now I have almost all [Pass] except for Cotton, BuildAll, Compile, and Post-Compile. Just wanna make sure, does this mean that I have succeeded in compiling it? (I don't need to simulate cotton, only focusing on maize now)
![image](https://github.com/APSIMInitiative/APSIMClassic/assets/132005487/8737bf41-65cb-44a2-b4a4-e5245e63df47)

**Comment by peter-devoil** [comment]
Yes, the "fail" cascades up the tree of targets - the cotton module is causing the others to fail.

Once upon a time, somebody decided that the IP of the cotton module was valuable and made us put a password on it..
